zerver/lib/streams.py

```diff
@@ -884,7 +896,7 @@ def filter_stream_authorization(
         if stream.id not in {stream.id for stream in unauthorized_streams}
         and stream.id not in {stream.id for stream in streams_to_which_user_cannot_add_subscribers}
     ]
-    return (
+    return StreamsCategorizedByPermissions(
         authorized_streams,
         unauthorized_streams,
         streams_to_which_user_cannot_add_subscribers,
```

Comment:
I know this works in JS, but I think in Python, this might be reliant on the fields happening to be in the same order as the dataclass definition?

Reply:
Fixed!